cmake/StaticBuild.cmake

```diff
@@ -58,6 +77,10 @@ if(BUILD_STATIC)
       set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -lcrypt32")
       # And sockets
       set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -lws2_32")
+      # link ncrypt and unistring statically
+      set(GNUTLS_LIBRARIES "${GNUTLS_LIBRARIES} -Wl,-Bstatic -lncrypt -Wl,-Bdynamic")
+      # p11-kit only available as dynamic library for MSYS2 on Windows and dynamic linking of unistring is required
```

Comment:
This is problematic as we don't want to force a p11-kit dependency on people that are trying to get a properly static build.

I think we need to better detect if p11-kit is required by GnuTLS in such a case. And really test if a static library is missing.

Reply:
I agree that would be more robust.  However, I don't know how to do that in CMAKE.  The current codebase doesn't compile for Windows without modification.  This will at least allow one setup to compile for Windows.  Unless you know of a setup that currently compiles without modification that this breaks?

Reply:
This breaks at least our Linux mingw builds. So it would cause problems for existing users.

Reply:
This doesn't highlight the p11-kit code.  However, I did make the p11-kit include dependent on finding the p11-kit library.  That is not the same as determining if GnuTLS requires p11-kit.  However, is that sufficient to prevent breaking your build?